.github/workflows/publish.yml

```diff
@@ -0,0 +1,23 @@
+name: Publish
```

Comment:
I moved this from Travis to a GH action for a few reasons:
- Mitigates vendor lock-in (i.e., we're more likely to stay on GitHub than on Travis)
- Encourages good etiquette on creating Releases (i.e., won't publish to npm without a release created)
- Officially supported Slack notification integrations
- I wanted to test out GH actions

Reply:
The fun (read: not so fun) bit about actions is you can't actually test them until they're merged. I added this to a private test repo on chawes13 and iterated a little bit so I'm _reasonably_ confident that this'll work as expected 🙂

Reply:
👌